overlord/snapshotstate/backend/backend.go

```diff
@@ -445,6 +457,118 @@ func addDirToZip(ctx context.Context, snapshot *client.Snapshot, w *zip.Writer,
 	return nil
 }
 
+// Import a snapshot from the export file format
+func Import(ctx context.Context, id uint64, r io.Reader) (int64, []string, error) {
+	comment := fmt.Sprintf("snapshot %d", id)
+
+	// prepare cache location to unpack the import file
+	p := path.Join(dirs.SnapCacheDir, "snapshots", fmt.Sprintf("import-%d", id))
+	if err := os.MkdirAll(path.Join(dirs.SnapCacheDir, "snapshots"), 0755); err != nil {
+		return 0, nil, fmt.Errorf("cannot create snapshots directory")
+	}
+	if err := os.Mkdir(p, 0755); err != nil {
+		if os.IsExist(err) {
+			return 0, nil, fmt.Errorf("snapshot import already in progress for ID `%d`", id)
+		}
+		return 0, nil, fmt.Errorf("failed creating import cache: %v", err)
+	}
+	defer os.RemoveAll(p)
+
+	// unpack the tar file to a temporary location (so the contents can be validated)
+	exportFound, size, err := unpackSnapshotImport(r, p)
+	if err != nil {
+		return 0, nil, err
+	}
+	if !exportFound {
+		return 0, nil, fmt.Errorf("snapshot import file incomplete: no export.json file")
+	}
+
+	// XXX: check the snapshot hashes (these are not in export.json at present)
+
+	// walk the cache directory to store the files
+	dir, err := osOpen(p)
+	if err != nil {
+		return 0, nil, fmt.Errorf("failed opening import cache: %v", comment)
+	}
+	defer dir.Close()
+	names, err := dirNames(dir, 100)
```

Comment:
This returns up to 100 names, but needs to be repeated until EOF.

Reply:
yes